src/lib.rs

```diff
@@ -411,11 +448,35 @@ impl Datadog {
         tags: impl IntoIterator<Item = String>,
     ) {
         if self.is_reporting_enabled {
-            self.client.event(
-                metric.as_ref(),
-                text.as_ref(),
-                tags.into_iter().collect::<Vec<String>>(),
-            );
+            let tags = tags.into_iter().collect::<Vec<String>>();
+            let metric = metric.as_ref();
+            self.track_tags(metric, &tags);
+            self.client.event(metric, text.as_ref(), tags);
+        }
+    }
+
+    fn track_tags<'a>(&self, metric: impl AsRef<str>, tags: impl IntoIterator<Item = &'a String>) {
+        let hashes = tags
+            .into_iter()
+            .map(|t| {
+                let mut hasher = DefaultHasher::new();
+                hasher.write(metric.as_ref().as_bytes());
+                hasher.write(t.as_bytes());
+                hasher.finish()
+            })
+            .collect::<Vec<u64>>();
+        let mut seen_tags = self.tag_tracker.seen_tags.lock().unwrap();
+        seen_tags.extend(hashes);
+        let seen_tag_count = seen_tags.len();
+        drop(seen_tags);
```

Comment:
I think the more idiomatic way to do this would be to create a block instead of calling drop, since they're easier to keep track of visually